attestation-agent/attestation-agent/src/lib.rs

```diff
@@ -154,17 +200,41 @@ impl AttestationAPIs for AttestationAgent {
         Ok(evidence.into_bytes())
     }
 
-    /// Extend runtime measurement register
+    /// Extend runtime measurement register. Parameters
+    /// - `events`: a event slice. Any single event will be calculated into a hash digest to extend the current
+    /// platform's RTMR.
+    /// - `register_index`: a target PCR that will be used to extend RTMR. Note that different platform
+    /// would have its own strategy to map a PCR index into a architectual RTMR index. If not given, a default one
+    /// will be used.
     async fn extend_runtime_measurement(
         &mut self,
-        events: Vec<Vec<u8>>,
+        domain: &str,
+        operation: &str,
+        content: &str,
         register_index: Option<u64>,
     ) -> Result<()> {
+        let register_index = register_index.unwrap_or_else(|| {
+            info!("No PCR index provided, use default {DEFAULT_PCR_INDEX}");
+            DEFAULT_PCR_INDEX
+        });
+
+        let log_entry = format!("{domain} {operation} {content}");
+        let event_digest = self
+            ._config
+            .eventlog_config
+            .eventlog_algorithm
+            .digest(log_entry.as_bytes());
```

Comment:
I think it would be good to move this transformation to a dedicated fn + unit_test, if just to specify it
```rust
struct Event<'a> {
  domain: &'a str,
  operation: &'a str,
  content: &'a str,
}

impl Event {
  fn log_digest(&self) -> Vec<u8> {
    let log_entry = format!("{domain} {operation} {content}");
    ...
  }
}
```

Reply:
I added `Event` struct, `to_string()` and `digest_with()` function.